services/MerchStoreService.ts

```diff
@@ -208,6 +212,33 @@ export default class MerchStoreService {
         });
       }
 
+      if (photos) {
+        // error on duplicate photo uuids
+        const dupSet = new Set();
+        photos.forEach((photo) => {
+          if (dupSet.has(photo.uuid)) { throw new UserError(`Multiple edits is made to photo: ${photo.uuid}`); }
+          dupSet.add(photo.uuid);
+        });
+
+        const photoUpdatesByUuid = new Map(photos.map((photo) => [photo.uuid, photo]));
+
+        item.photos.map((currentPhoto) => {
+          if (!photoUpdatesByUuid.has(currentPhoto.uuid)) return;
+          const photoUpdate = photoUpdatesByUuid.get(currentPhoto.uuid);
+          // photo positions are 0-index
+          return MerchandiseItemPhotoModel.merge(currentPhoto, photoUpdate);
+        });
+
```

Comment:
This is a little confusing to me. It just creates a map of uuid to new photos, then changes the old photo with that uuid to the new photo right?

Reply:
Specifically, this part handles updating the positions of the pictures. The merge will propagate this change to the database.